### PR TITLE
Upgrade Jest to 22.1 and dedupe packages

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

This PR updates to Jest 22.1 which give our tests better memory usage and adds interactive snapshots. Link to full changelog [here](https://github.com/facebook/jest/blob/master/CHANGELOG.md#jest-2210)

**Update** this PR also dedupes packages using [yarn-tools](https://www.npmjs.com/package/yarn-tools)

### CODE REVIEW

**Comment by gaearon** [comment]
Need to update babel-jest too?

**Comment by rickhanlonii** [comment]
Good catch, updated

File: yarn.lock [hunk 1]
```diff
@@ -10,6 +10,14 @@
     esutils "^2.0.2"
     js-tokens "^3.0.0"
 
+"@babel/code-frame@^7.0.0-beta.35":
```
Comment (SimenB) [inline yarn.lock]: you should dedupe

**Comment by rickhanlonii** [comment]
Updates:
- Removes babel-jest (Jest will install automatically)
- Dedupes the packages in yarn.lock (shout out @SimenB for the assist)

**Comment by SimenB** [comment]
@rickhanlonii mind adding a dependency on `expect`? The reason it doesn't crash now is because of hoisting, see https://github.com/facebook/react/blob/4ca7855ca062d5d7dfca83c86acf46731e1e57ef/packages/react-noop-renderer/src/ReactNoop.js#L25

I also think `babel-jest` should be here, it needs to be hoisted to the top. It currently works because of an implementation detail of yarn (we `require` it in Jest at the top level, so if it ends up being inside of `jest` (not hoisted) we get the wrong one https://github.com/facebook/jest/blob/36c705bb1eac380e0926487bbe8b8db0aa2fa2e8/packages/jest-config/src/normalize.js#L96-L128)

**Comment by rickhanlonii** [comment]
@SimenB what do you think about removing the expect import instead? That appears to be the only usage and doesn't seem to be necessary?

**Comment by SimenB** [comment]
No feelings either way, up to the React maintainers 🙂 But the removal was me jumping the gun at #12008

**Comment by rickhanlonii** [comment]
Good bot

Going to close until I can sort that out

**Pull request closed without merging**